BonApp.js

```diff
@@ -47,306 +45,275 @@ render: function(output) {
-        var theValue = JSON.parse(savedDiet).diet;
-        document.querySelector('.myDiet').value = theValue;
+        savedDiet = localStorage.getItem('BonAppSettings')
+        var theValue = JSON.parse(savedDiet).diet
+        document.querySelector('.myDiet').value = theValue
       }
       else {
-        return -1;
+        return -1
       }
 
-      return theValue;
-    };
-
+      return theValue
+    }
 
-    dietSelection.onchange = function() {
-      return saveSettings(dietSelection.value);
-    };
-
-
-    var switchNum = 0;
+    var switchNum = 0
 
     var changeTheView = function() {
 
       if (switchNum % 2 == 0) {
-        $('#cafeSelection').show();
-        $('#main').hide();
-        $("#spinner").hide();
-        $('#topRightIcon').css('background-image', 'url(BonApp-Widget/img/arrow.png)');
-        $('#bonApp').css('height', '200px');
-        $('.queryBox').focus();
-
-        initSearch();
+        $('#cafeSelection').show()
+        $('#main').hide()
+        $("#spinner").hide()
+        $('#topRightIcon').css('background-image', 'url(BonApp-Widget/img/arrow.png)')
+        $('#bonApp').css('height', '200px')
+        $('.queryBox').focus()
+
+        initSearch()
       }
       else {
-        $('#cafeSelection').hide();
-        $('#main').show();
-        $('#topRightIcon').css('background-image', 'url(BonApp-Widget/img/search.png)');
-        $('#bonApp').css('height', 'auto');
+        $('#cafeSelection').hide()
+        $('#main').show()
+        $('#topRightIcon').css('background-image', 'url(BonApp-Widget/img/search.png)')
+        $('#bonApp').css('height', 'auto')
       }
 
-      return switchNum++;
-    };
+      return switchNum++
+    }
 
 
     document.getElementById('topRightIcon').addEventListener('click', function() {
-      return changeTheView();
-    });
+      return changeTheView()
+    })
 
 
    var makeItAllHappen = function(output) {
-      var dom          = $(domEl);
-      var savedDiet    = initFromLocalStorage();
-      var jstringObj   = JSON.parse(output);
-      var title        = parseCafe(jstringObj);
-      var theDate      = parseDate(jstringObj.days[0].date);
-      var theMenu      = jstringObj.items;
-      var friendly     = parseMenu(theMenu, savedDiet);
-      var outputString = createOutputString(friendly);
-      var widgetSelect = 'Find Your Café';
+      var savedDiet    = initFromLocalStorage()
+      var jstringObj   = JSON.parse(output)
+      var title        = parseCafe(jstringObj)
+      var theDate      = parseDate(jstringObj.days[0].date)
+      var theMenu      = jstringObj.items
+      var friendly     = parseMenu(theMenu, savedDiet)
+      var outputString = createOutputString(friendly)
+      var widgetSelect = 'Find Your Café'
+
+      dietSelection.onchange = function() {
+        return saveSettings(dietSelection.value)
+      }
+
+      var saveSettings = function(diet) {
+        var friendly, userSettings
```

Comment:
Define variables where they're actually needed, instead of at the top of the function, please.